cloud/cloudstack/cs_instance.py

```diff
@@ -448,9 +464,25 @@ def get_instance(self):
                         break
         return self.instance
 
+    def get_iptonetwork_mappings(self):
+        network_mappings = self.module.params.get('iptonetwork')
+        if network_mappings is None:
+            return
+
+        if network_mappings and self.module.params.get('networks'):
+            self.module.fail_json(msg="networks and iptonetwork are mutually exclusive.")
+
+        network_names = [n['network'] for n in network_mappings]
+        ids = self.get_network_ids(network_names).split(',')
```

Comment:
we should change `get_network_ids()` to return a list and make the `','.join()` where we use `self.get_network_ids()`. This way we do not need to split here again.